screen/settings/src/main/java/com/ivy/settings/SettingsScreen.kt

```diff
@@ -160,14 +166,17 @@ private fun BoxWithConstraintsScope.UI(
     onSetStartDateOfMonth: (Int) -> Unit = {},
     onDeleteAllUserData: () -> Unit = {},
     onDeleteCloudUserData: () -> Unit = {},
-
-    ) {
+    onSwitchLanguage: () -> Unit = {}
+) {
     var currencyModalVisible by remember { mutableStateOf(false) }
     var nameModalVisible by remember { mutableStateOf(false) }
     var chooseStartDateOfMonthVisible by remember { mutableStateOf(false) }
     var deleteCloudDataModalVisible by remember { mutableStateOf(false) }
     var deleteAllDataModalVisible by remember { mutableStateOf(false) }
     var deleteAllDataModalFinalVisible by remember { mutableStateOf(false) }
+    val languageOptionVisible by remember {
+        mutableStateOf(Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU)
```

Comment:
This can be just a regular boolean, no need to be a state

Reply:
Also, let's move this logic inside the ViewModel and add `languageOptionVisible` to the view-state / ui state data model

Reply:
ok

Reply:
Is it a good idea to have a single UI state? because any change to that data class will recompose all the components that depends on it.

Reply:
Yeah, that's perfectly fine - it's recommended and all big tech use it, check the developer guidelines referencs. The reason: Compose is smart enough to recompose only the UI that needs an update. When you give Compose an immutable (stable) state it can pick only the nodes that needs update and recompose only those composables

Reply:
We also have a CI check that verifies that https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md#compose-stability